db/multi.go

```diff
@@ -10,37 +10,56 @@ import (
 // query which inserts multiple rows into the same table. It can also execute
 // the resulting query.
 type MultiInserter struct {
-	table     string
-	fields    string
-	retCol    string
-	numFields int
-	values    [][]interface{}
+	// These are validated by the constructor as containing only characters
+	// that are allowed in an unquoted identifier.
+	// https://mariadb.com/kb/en/identifier-names/#unquoted
+	table           string
+	fields          []string
+	returningColumn string
+
+	values [][]interface{}
 }
 
 // NewMultiInserter creates a new MultiInserter, checking for reasonable table
-// name and list of fields.
-func NewMultiInserter(table string, fields string, retCol string) (*MultiInserter, error) {
-	numFields := len(strings.Split(fields, ","))
-	if len(table) == 0 || len(fields) == 0 || numFields == 0 {
+// name and list of fields. returningColumn is the name of a column to be used
+// in a `RETURNING xyz` clause at the end. If it is empty, no `RETURNING xyz`
+// clause is used. If returningColumn is present, it must refer to a column
+// that can be parsed into an int64.
+// Safety: `table`, `fields`, and `returningColumn` must contain only strings
```

Comment:
Heads-up to other reviewers that this will have a merge conflict with https://github.com/letsencrypt/boulder/pull/6568 but that's fine and expected.